contrib/mesos/pkg/scheduler/podtask/pod_task.go

```diff
@@ -267,9 +273,10 @@ func RecoverFrom(pod api.Pod) (*T, bool, error) {
 		podKey:     key,
 		State:      StatePending, // possibly running? mesos will tell us during reconciliation
 		Flags:      make(map[FlagType]struct{}),
-		mapper:     MappingTypeForPod(&pod),
+		mapper:     NewHostPortMapper(&pod),
 		launchTime: now,
 		bindTime:   now,
+		Spec:       &Spec{},
```

Comment:
We were actually discussing this. The Spec != nil means that the pod is scheduled. This can also be signaled with a simple flag.